fence/agents/powerman/fence_powerman.py

```
@@ -0,0 +1,300 @@
+
+	def __init__(self, server_name, port="10101"):
+		"""
+		Args:
+			server_name: (string) host or ip of powerman server
+			port: (str) port number that the powerman server is listening on
+		"""
+		self.server_name = server_name
+		self.port = port
+		self.server_and_port = server_name + ":" + str(port)
+		# this is a list of the command and its options. For example:
+		# ['powerman', '--server-host', 'elssd1:10101']
+		self.base_cmd = [
+			self.program_name, 
+			"--server-host", 
+			self.server_and_port
+		]
+
+	def _run(self, cmd, only_first_line):
+		# Args:
+		#   cmd: (list) commands and arguments to pass to the program_name
+
+		# add the 2 command lists together to get whole command to run
+		run_this = self.base_cmd + cmd 
+		try:
+			popen = subprocess.Popen(run_this, stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+			out = popen.communicate()
+		except OSError as e:
+			logging.error("_run command error: %s\n", e)
+			sys.exit(1)
+		if only_first_line == True:
+			result_line = out[0].decode().strip()
+			return (result_line, popen.returncode)
+		else:
+			result_list = []
+			for line in out:
+				result_list.append(line)
+			return (result_list, popen.returncode)
+
+	def is_running(self):
+		"""simple query to see if powerman server is responding. Returns boolean"""
+		cmd = ["-q"] # just check if we get a response from the server
+		result, ret_code = self._run(cmd, True)
+		if ret_code != 0:
+			return False
+		return True
+
+	def on(self, host):
+		logging.debug("PowerMan on: %s\n", host)
+		cmd = ["--on", host]
+		try:
+			result, ret_code = self._run(cmd, True)
+		except OSError as e:
+			logging.error("PowerMan Error: The command '--on' failed: %s\n", e)
+			return -1
+		except ValueError as e:
+			logging.error("PowerMan Error: Popen: invalid arguments: %s\n", e)
+			return -1
+		logging.debug("pm.on result: %s ret_code: %s\n", result, ret_code)
+
+		## For some devices, the "on" or "off" command can fail silently,
```

Do you mean I don't need this block of code:
`
                ## For some devices, the "on" or "off" command can fail silently,
                ## but for query powerman keeps trying until it gets a valid answer;
                ## so we must check after trying to power on/off a device.
                ## Powerman seems to report the old state if asked too quickly, wait a second.

                time.sleep(1)
                queryret = self.query(host)

                if (ret_code == 0 and queryret == "off") or (ret_code < 0 and queryret == "on"):
                        logging.warning("command '%s' returned (%s) but query reported (%s) afterwards\n", cmd, result, queryret)

                if queryret == "on":
                        return 0
                else:
                        return -1
`

and should just return ret_code from the "on" or "off" commands?